tests/src/test/scala/com/nvidia/spark/rapids/timezone/TimeZonePerfSuite.scala

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.timezone
+
+import java.io.File
+import java.time.Instant
+
+import com.nvidia.spark.rapids.SparkQueryCompareTestSuite
+import com.nvidia.spark.rapids.jni.GpuTimeZoneDB
+import org.scalatest.BeforeAndAfterAll
+
+import org.apache.spark.SparkConf
+import org.apache.spark.sql._
+import org.apache.spark.sql.functions._
+import org.apache.spark.sql.timezone._
+import org.apache.spark.sql.timezone.TimeZonePerfUtils
+import org.apache.spark.sql.types._
+
+/**
+ * A simple test performance framework for non-UTC features.
+ * Usage:
+ *   update enablePerfTest = true in this code
+ *   mvn test -Dbuildver=311 -DwildcardSuites=com.nvidia.spark.rapids.timezone.TimeZonePerfSuite
+ * Note:
+ *   Generate a Parquet file with 6 columns:
+ *     - c_ts: timestamp column
+ *     - c_long_of_ts: long value which is microseconds
+ *     - c_date: date column
+ *     - c_int_of_date:int value which is days from 1970-01-01
+ *     - c_str_for_cast: strings for cast to timestamp, formats are yyyy, yyyy-mm, ...
+ *     - c_str_of_ts: strings with format: yyyy-MM-dd HH:mm:ss
+ *    Each column is high duplicated.
+ *    The generated file is highly compressed since we expect both CPU and GPU can scan quickly.
+ *    When testing operators, we need to add in a max/count aggregator to reduce the result data.
+ */
+class TimeZonePerfSuite extends SparkQueryCompareTestSuite with BeforeAndAfterAll {
+  // perf test is disabled by default since it's a long running time in UT.
+  private val enablePerfTest = true
+
+  // rows for perf test
+  private val numRows: Long = 1024L * 1024L * 10L
+
+  private val zones = Seq(
+    "Iran"
+    // "America/Los_Angeles" // DST is not supported now
+  )
+
+  private val path = "/tmp/tmp_TimeZonePerfSuite"
+
+  /**
+   * Create a Parquet file to test
+   */
+  override def beforeAll(): Unit = {
+    withCpuSparkSession(
+      spark => createDF(spark).write.mode("overwrite").parquet(path))
+  }
+
+  override def afterAll(): Unit = {
+    FileUtils.deleteRecursively(new File(path))
+  }
+
+  val year1980 = Instant.parse("1980-01-01T00:00:00Z").getEpochSecond * 1000L * 1000L
+  val year2000 = Instant.parse("2000-01-01T00:00:00Z").getEpochSecond * 1000L * 1000L
+  val year2030 = Instant.parse("2030-01-01T00:00:00Z").getEpochSecond * 1000L * 1000L
+
+  private val stringsForCast = Array(
```

Comment:
Shall we put every zone IDs into this pool by loading them from `TimeZoneDB` ?  Either, appends more zones into this pool.

Reply:
Now we can specify the time zones:
```
-DTZs=Iran,Asia/Shanghai -DenableTimeZonePerf=true
```